game-core/src/main/java/games/strategy/triplea/settings/ClientSetting.java

```diff
@@ -162,23 +162,22 @@ static void resetPreferences() {
   }
 
   @Override
-  public final void addSaveListener(final Consumer<GameSetting<T>> saveListener) {
-    Preconditions.checkNotNull(saveListener);
-    onSaveActions.add(saveListener);
+  public final void addListener(final Consumer<GameSetting<T>> listener) {
+    Preconditions.checkNotNull(listener);
+    listeners.add(listener);
   }
 
   @Override
-  public final void removeSaveListener(final Consumer<GameSetting<T>> saveListener) {
-    onSaveActions.remove(saveListener);
+  public final void removeListener(final Consumer<GameSetting<T>> listener) {
+    listeners.remove(listener);
   }
 
   public static void showSettingsWindow() {
     SettingsWindow.INSTANCE.open();
   }
 
   /**
-   * Persists user preferences.x Note: 'value()' read calls will not pick up any new
-   * values saved values until after 'flush' has been called.
```

Comment:
I removed this comment because it doesn't appear to be true.  I verified that calling `setValue()` followed by `getValue()`, without an intervening `flush()`, returns the value passed to `setValue()`.  I believe the Java preferences subsystem caches the preference values in memory and doesn't always read them from the backing store.

In a previous PR, I observed that, even if we don't explicitly flush the preferences, the JVM implicitly flushes them when it is shut down.  So, unless it is required that the backing store be updated in real time with the in-memory cache, we might even be able to get rid of the entire concept of _flushing_ from `ClientSetting`.